app/controllers/api/v1/comments_controller.rb

```diff
@@ -1,55 +1,24 @@
 class Api::V1::CommentsController < ApplicationController
-    before_action :set_product
-    before_action :set_comment, only: [:show, :update, :destroy]
+  before_action :set_product, only: [:product_comment]
 
-    def index
-        @comments = Comment.all
-        render json: @comments
-    end
-  
-    def show
-        render json: @comment
-    end
-  
-    def create
-      @comment = @product.comments.build(comment_params)
-  
-      if @comment.save
-        render json: @comment, status: 201
-      else
-        render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
-      end
+  def product_comment
+    @comment = @product.comments.new(comment_params)
+    @comment.commentable = @product
+    if @comment.save
+      render json: @comment, status: 201
+    else
+      render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
     end
+  end
 
-    def update
-      if @comment.update(comment_params)
-        render json: @comment, status: :ok
-      else
-        render json: { errors: @comment.errors.full_messages }, status: :unprocessable_entity
-      end
-    end
-  
-    def destroy
-      if  @comment.destroy
-        render json: nil, status: :no_content
-      else
-        render json: { errors: @comment.errors }, status: :unprocessable_entity
-      end
-    end
+  private
 
-    private
+    def set_product
+      @product = Product.find(params[:product_id])
```

Comment:
What happens if params didn't have product_id? An exception should be raised but where are we handling that exception?